File: JAVA DSA SHEET/Problems_On_Arrays/TwoSum.java
package Problems_On_Arrays;
import java.util.*;

public class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int result[] = new int[2];
        for(int i = 0; i < nums.length; i++) {
            int temp = target - nums[i];
            if(map.containsKey(temp)) {
                result[0] = map.get(temp);
                result[1] = i;
                return result;
            }
            map.put(nums[i], i);
        }
        return new int[]{-1, -1};
    }

    public static void main(String[] args) {
        TwoSum obj = new TwoSum();
        int[] nums = {2, 7, 11, 15};
        int target = 9;
        int[] res = obj.twoSum(nums, target);
        System.out.println(Arrays.toString(res));
    }
}

File: JAVASCRIPT PROGRAMS/TwoSum.js
function twoSum(nums, target) {
    const map = new Map();

    for (let i = 0; i < nums.length; i++) {
        let diff = target - nums[i];

        if (map.has(diff)) {
            return [map.get(diff), i];
        }

        map.set(nums[i], i);
    }
}

let nums = [10, 5, 2, 15, 20];
let target = 7;

let result = twoSum(nums, target);
console.log(result);